# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/scene_building.rs
# Commit: 0919633e25e9
# Full Hash: 0919633e25e99241c2eb06374ae2f54d31ae132d
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2021-04-22 21:31:57
# Regressor Bug: 1702228
# File Overlap Count: 1
# Description:
#   Bug 1702228 - Check gradient parameters. r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112969
# ==============================================================================

diff -r e38dce4a5b06 -r 0919633e25e9 gfx/wr/webrender/src/scene_building.rs
--- a/gfx/wr/webrender/src/scene_building.rs	Thu Apr 22 10:34:54 2021 +0000
+++ b/gfx/wr/webrender/src/scene_building.rs	Thu Apr 22 10:34:55 2021 +0000
@@ -1215,6 +1215,10 @@
             DisplayItem::Gradient(ref info) => {
                 profile_scope!("gradient");
 
+                if !info.gradient.is_valid() {
+                    return;
+                }
+
                 let (mut layout, unsnapped_rect, spatial_node_index, clip_chain_id) = self.process_common_properties_with_bounds(
                     &info.common,
                     &info.bounds,
@@ -1287,6 +1291,10 @@
             DisplayItem::RadialGradient(ref info) => {
                 profile_scope!("radial");
 
+                if !info.gradient.is_valid() {
+                    return;
+                }
+
                 let (mut layout, unsnapped_rect, spatial_node_index, clip_chain_id) = self.process_common_properties_with_bounds(
                     &info.common,
                     &info.bounds,
@@ -1360,6 +1368,10 @@
             DisplayItem::ConicGradient(ref info) => {
                 profile_scope!("conic");
 
+                if !info.gradient.is_valid() {
+                    return;
+                }
+
                 let (mut layout, unsnapped_rect, spatial_node_index, clip_chain_id) = self.process_common_properties_with_bounds(
                     &info.common,
                     &info.bounds,